### PR TITLE
Azure DNS Resolver Private Preview 

### PR DESCRIPTION
**Pull request by guandali**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
converted to draft since no review needed.

File: src/DnsResolver/custom/readme.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+- Proxy
+- ProxyCredential
+- ProxyUseDefaultCredentials
+
+These provide functionality to our HTTP pipeline and other useful features. In script, you can forward these parameters using `$PSBoundParameters` to the other cmdlets you're calling within `Az.DnsResolver`. For C#, follow the usage seen in the `ProcessRecordAsync` method.
+
+### Attributes
+For processing the cmdlets, we've created some additional attributes:
+- `Microsoft.Azure.PowerShell.Cmdlets.DnsResolver.DescriptionAttribute`
+  - Used in C# cmdlets to provide a high-level description of the cmdlet. This is propegated to reference documentation via [help comments](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comment_based_help) in the exported scripts.
```
Comment (wyunchi-ms) [inline src/DnsResolver/custom/readme.md:35]: We will remove the language info in url.
Please update `autorest-powershell` in your environment and regenerate these files.
``` powershell
autorest --reset
autorest
```

File: src/DnsResolver/readme.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+> see https://aka.ms/autorest
+
+``` yaml
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file:
+  - C:\Networking\DNS\azure-rest-api-specs-pr\specification\dnsresolver\resource-manager\Microsoft.Network\preview\2020-04-01-preview\dnsresolver.json
+
+module-version: 0.1.0
+title: DnsResolver
```
Comment (wyunchi-ms) [inline src/DnsResolver/readme.md:77]: please add latest commit id of swagger as the value of `branch`. You can find example [here](https://github.com/Azure/azure-powershell/blob/wyunchi%2Fgen2master%2FCloudService/src/CloudService/readme.md#L50).

Comment (guandali) [reply src/DnsResolver/readme.md:80]: Updated. 
Note that our swagger in on azure-rest-api-specs-pr

File: src/DnsResolver/readme.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+---
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+branch: ffebe2759a66ebd6ca52288a9eaf1c02f28e3842
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file:
+  - C:\Networking\DNS\azure-rest-api-specs-pr\specification\dnsresolver\resource-manager\Microsoft.Network\preview\2020-04-01-preview\dnsresolver.json
```
Comment (wyunchi-ms) [inline src/DnsResolver/readme.md:77]: Please don't use local swagger file. You need to use a swagger file in the publish repo.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**